src/main/scala/battleship/GameActor.scala

```diff
@@ -0,0 +1,111 @@
+package battleship
+
+import akka.actor.{Actor, ActorLogging, ActorRef, Props}
+import battleship.GameActor._
+import battleship.PlayerActor.{GetNextShot, PlaceBoats}
+import battleship.game._
+
+/**
+  * Created by m06f947 on 3-8-2017.
+  */
+
+
+
+object GameActor {
+
+  case class StartGame(boardsize:Int, player1:ActorRef, player2:ActorRef, boatset: Seq[Boat])
+
+  //case class WhatIsBoatSetup(boats:Seq[Boat], boardsize: Int)
+  case class ThisIsBoatSetup(placement: Set[(Boat,BoatLocation)])
+  //case class WhatIsNextMove(boardsize:Int, shotHistory: Seq[((Int,Int),ShotResult)])
+  case class ThisIsNextMove(x: Int, y: Int)
+
+
+  def props: Props = Props(new GameActor)
+
+}
+
+class GameActor extends Actor with ActorLogging {
+
+
+  //internal state
+  var boardsize = Game.defaultBoardSize
+  var boatSet = Game.defaultBoatSet
+  //first of pair is currentplayer
+  var players: (ActorRef, ActorRef) = (Actor.noSender, Actor.noSender)
+  def currentPlayer = players._1
+  def swapPlayers = {players = (players._2, players._1)}
+  //Have the boatsetup ready. In here player1 is boatsetup._1
+  var boatsetup: Map[ActorRef, Set[(Boat, BoatLocation)]] = Map()
+  //keep the state of the boards
+  var boardstates: Map[ActorRef, BoardState] = Map()
+
+  def initPlayerState(playerToPlaceBoats: ActorRef): BoardState = {
+    val placements = boatsetup(playerToPlaceBoats)
+    placements.foldLeft(BoardState.empty) {
+      case (boardState, (boat, location)) => boardState.placeBoat(boat, location)
+    }
+  }
+
+  def numberofboatsetupsreceived:Int = {boatsetup.count((a: (ActorRef, Set[(Boat, BoatLocation)])) => true)}
+
+  //*****************************************************************************************************************
+  //This is not functional programming at all, these functions are there only for the side effects...****************

```

Comment:
In order to take a more functional approach, you can take a look at `context.become` [here](http://doc.akka.io/docs/akka/current/scala/actors.html#become-unbecome). This was not yet treated while we discussed the basics of Akka, but you can use this to manage the actor state in a more functional way. Note that you can give parameters to the custom state, as in
`override def receive: Receive = {case anyMessage => context.become(customState(12)) }`
`def customState(param: Int): Receive = {case receivedInteger: Int => context.become(customState(param + receivedInteger)) }`